examples/gpgpu.rs

```diff
@@ -6,28 +6,34 @@ use glium::glutin;
 fn main() {
     let context = glutin::HeadlessRendererBuilder::new(1024, 1024).build().unwrap();
     let display = glium::HeadlessRenderer::new(context).unwrap();
-
+        
     let program = glium::program::ComputeShader::from_source(&display, r#"\
 
             #version 430
             layout(local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
 
             layout(std140) buffer MyBlock {
                 float power;
-                float values[256];
+                vec4 values[4096/4];
             };
 
             void main() {
-                float val = values[gl_GlobalInvocationID.x];
-                values[gl_GlobalInvocationID.x] = pow(val, power);
+                vec4 val = values[gl_GlobalInvocationID.x];
+                
+                values[gl_GlobalInvocationID.x] = vec4(
```

Comment:
No need to change this line. `pow(val, power);` should work as well.

Reply:
I didn't think of that! However, now that I tried it it seems it doesn't actually work. The glsl pow() function is only defined for scalars. At least my nvidia 970 drivers on linux don't accept a vec4 as first parameter.

Reply:
Or maybe both arguments need to be vec4. Didn't try that. Sorry for speaking too quickly!